### PR TITLE
adapt to coq/coq#15220

### PR DESCRIPTION
**Pull request by gares**
To be merge when https://github.com/coq/coq/pull/15220 is.

### CODE REVIEW

File: _CoqProject.in [hunk 1]
```diff
@@ -1,3 +1,4 @@
+META.coq-fiat-parsers
```
Comment (JasonGross) [inline _CoqProject.in]: This is not backwards compatible.  Instead the Makefile should be updated to insert this line only on versions of Coq new enough to support it

**Review by JasonGross — CHANGES_REQUESTED** [review]

File: _CoqProject.in (hunk 1 above)
Comment (gares) [reply _CoqProject.in]: can you point me to the piece of code you want to me patch? Do you already test the version of Coq?

Comment (JasonGross) [reply _CoqProject.in]: You should add an `echo` to
https://github.com/mit-plv/fiat/blob/cb85ee76ee3410f04333633f6644663ebb525ac8/Makefile#L376-L377
in addition to adding it here, and then modify the code starting at https://github.com/mit-plv/fiat/blob/cb85ee76ee3410f04333633f6644663ebb525ac8/Makefile#L54 to extend `EXTRA_PIPE_SED_FOR_COQPROJECT ` at
https://github.com/mit-plv/fiat/blob/cb85ee76ee3410f04333633f6644663ebb525ac8/Makefile#L50
with something like `+= | sed 's/^META.*$$//g'` in old versions of Coq.

Comment (gares) [reply _CoqProject.in]: done.

**Comment by gares** [comment]
I believe the two .v files should be versioned as well, but I could not find any code doing it (for other files)

File: Makefile [hunk 2]
```diff
@@ -55,77 +55,90 @@ ifneq (,$(filter 8.4%,$(COQ_VERSION))) # 8.4 - this is a kludge to get around th
...
 OTHERFLAGS += -w "-deprecated-appcontext -notation-overridden"
 ML4_OR_MLG := mlg
+EXTRA_PIPE_SED_FOR_COQPROJECT += | set 's/@META@//g'
 else
 ifneq (,$(filter 8.15%,$(COQ_VERSION)))
 EXPECTED_EXT:=.v815
 ML_DESCRIPTION := "Coq v8.15"
 OTHERFLAGS += -w "-notation-overridden"
 ML4_OR_MLG := mlg
+EXTRA_PIPE_SED_FOR_COQPROJECT += | set 's/@META@//g'
```
Comment (JasonGross) [inline Makefile:134]: There's another v815 block a couple lines down which should be turned into v816 and get a pipe line like this

**Review by JasonGross — CHANGES_REQUESTED** [review]
You can version the .v files in exactly the same way the ml files are versioned, just add them to
https://github.com/mit-plv/fiat/blob/cb85ee76ee3410f04333633f6644663ebb525ac8/Makefile#L160
and add lines like
https://github.com/mit-plv/fiat/blob/cb85ee76ee3410f04333633f6644663ebb525ac8/Makefile#L384
and ignore the fact that all the variables say ML, it's not actually specific to ML files

Thanks for doing all of this!

File: Makefile (hunk 2 above)
Comment (gares) [reply Makefile:134]: done

**Comment by JasonGross** [comment]
It now builds on all versions.  Should I merge now, or does the fact that it builds on master mean that there's some mistake?

**Comment by gares** [comment]
The PR on the coq side is not final. Please wait.

**Comment by SkySkimmer** [comment]
Please merge now

**Pull request merged**